### PR TITLE
ab#75608 - Gridster

### PR DESCRIPTION
**Pull request by AntoineRelief**
# Description

Change widget grid system, to use gridster instead of kendo layout tiles.

## Useful links

- [ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/75608)
- [Package doc](https://www.npmjs.com/package/angular-gridster2)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (refactor or addition to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [ ] Test A
- [ ] Test B

## Screenshots

Please include screenshots of this change. If this issue is only back-end related, and does not involve any visual change of the platform, you can skip this part.

# Checklist:

( * == Mandatory ) 

- [ ] * I have set myself as assignee of the pull request
- [ ] * My code follows the style guidelines of this project
- [ ] * Linting does not generate new warnings
- [ ] * I have performed a self-review of my own code
- [ ] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [ ] * I have commented my code, particularly in hard-to-understand areas
- [ ] * I have put JSDoc comment in all required places
- [ ] * My changes generate no new warnings
- [ ] * I have included screenshots describing my changes if relevant
- [ ] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AntoineRelief** [comment]
Missing points:
- currently, moving & resizing widgets do not save the dashboard. There is a built way to trigger callbacks for that with gridster, but this way, it's calling the method too many times. I started to build a subject that would use debounceTime to prevent that, but it won't detect when changes are coming from the user, or from the resize event of the component itself
- how to improve it so existing dashboards can load fine. Unai built a migration in the back. We can certainly also build something in the front, so the front accepts both widgets definition ( using cols & rows, or defaultCols & defaultRows ).
- if all that is done, you can also check comments on the ticket ( last one, with proposed changes )

**Comment by unai-reliefapp** [comment]
Hi @AntoineRelief , pushed fixes for the points highlighted in here, plus also disable the drag event for widget actions, but still allowed from the widget header. I do not change the grid so admin can move widgets as pleased, I'll leave that to further editions if needed.
You can start checking if it's fine, the comments in the ticket would be just to call the [setGridOptions](https://github.com/ReliefApplications/ems-frontend/blob/gridster/libs/shared/src/lib/components/widget-grid/widget-grid.component.ts#L204) method on margin value is set from the dialog and use that value instead of the fixed one in there

KR, Unai

File: libs/shared/src/lib/components/widget-grid/widget-grid.component.ts [hunk 1]
```diff
@@ -134,6 +187,53 @@ export class WidgetGridComponent
...
+      margin: 10,
+      outerMargin: false,
+      minItemCols: 1, // min item number of columns
+      minItemRows: 1, // min item number of rows
+      maxCols: this.colsNumber,
+      minCols: this.colsNumber,
+      fixedRowHeight: 200,
+      draggable: {
+        enabled: this.canUpdate,
+        ignoreContentClass: 'widget-actions',
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/widget-grid/widget-grid.component.ts]: @unai-reliefapp 
if you do that you can also drag / drop from the content of the widget 😅

Comment (unai-reliefapp) [reply libs/shared/src/lib/components/widget-grid/widget-grid.component.ts]: uooh, that passed through 👀 😅 , thanks Antoine

**Pull request merged**